staging/src/k8s.io/client-go/transport/token_source_test.go

```diff
@@ -156,6 +156,68 @@ func TestCachingTokenSourceRace(t *testing.T) {
 	}
 }
 
+func TestTokenSourceTransportRoundTrip(t *testing.T) {
+	goodToken := &oauth2.Token{
+		AccessToken: "good",
+		Expiry:      time.Now().Add(1000 * time.Hour),
+	}
+	badToken := &oauth2.Token{
+		AccessToken: "bad",
+		Expiry:      time.Now().Add(1000 * time.Hour),
+	}
+	tests := []struct {
+		name        string
+		header      http.Header
+		token       *oauth2.Token
+		cachedToken *oauth2.Token
+		wantCalls   int
+	}{
+		{
+			name:   "skip oauth rt if has authorization header",
+			header: map[string][]string{"Authorization": {"Bearer TOKEN"}},
+			token:  goodToken,
+		},
+		{
+			name:      "authorized on newly acquired good token",
+			token:     goodToken,
+			wantCalls: 1,
+		},
+		{
+			name:        "authorized on cached good token",
+			token:       goodToken,
+			cachedToken: goodToken,
+			wantCalls:   0,
+		},
+		{
+			name:      "unauthorized on newly acquired bad token",
+			token:     badToken,
+			wantCalls: 1,
+		},
+		{
+			name:        "unauthorized on cached bad token",
+			token:       badToken,
+			cachedToken: badToken,
+			wantCalls:   0,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			tts := &testTokenSource{
+				tok: test.token,
+			}
+			cachedTokenSource := NewCachedTokenSource(tts)
+			cachedTokenSource.tok = test.cachedToken
+
+			rt := ResettableTokenSourceWrapTransport(cachedTokenSource)(&testTransport{})
+
+			rt.RoundTrip(&http.Request{Header: test.header})
+			if tts.calls != test.wantCalls {
+				t.Errorf("RoundTrip() called Token() = %d times, want %d", tts.calls, test.wantCalls)
+			}
+		})
```

Comment:
add an assertion to the test about whether or not we expect to have a cached token after making the call, and check it here

Reply:
done